docs/elasticsearch/search-template.md

```diff
@@ -0,0 +1,416 @@
+```json
+GET _search/template
+{
+  "id": "play_search_template",
+  "params": {
+    "play_name": "Henry IV",
+    "page": 0,
+    "size": 1
+  }
+}
+```
+#### Sample output
+
+```json
+{
+  "took": 7,
+  "timed_out": false,
+  "_shards": {
+    "total": 6,
+    "successful": 6,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 3205,
+      "relation": "eq"
+    },
+    "max_score": 3.641852,
+    "hits": [
+      {
+        "_index": "shakespeare",
+        "_type": "_doc",
+        "_id": "4",
+        "_score": 3.641852,
+        "_source": {
+          "type": "line",
+          "line_id": 5,
+          "play_name": "Henry IV",
+          "speech_number": 1,
+          "line_number": "1.1.2",
+          "speaker": "KING HENRY IV",
+          "text_entry": "Find we a time for frighted peace to pant,"
+        }
+      }
+    ]
+  }
+}
+```
+
+If you have a stored template and want to validate it, you can use the `render` operation:
+
+```json
+POST _render/template
+{
+  "id": "play_search_template",
+  "params": {
+    "play_name": "Henry IV"
+  }
+}
+```
+
+#### Sample output
+
+```json
+{
+  "template_output": {
+    "from": "0",
+    "size": "10",
+    "query": {
+      "match": {
+        "play_name": "Henry IV"
+      }
+    }
+  }
+}
+```
+
+## Advanced parameter conversion with search templates
+
+You have a lot of different syntax options in Mustache to transpose the input parameters into a query.
+You can specify conditions, run loops, join arrays, convert arrays to JSON, and so on.
+
+### Conditions
+
+You can use the section tag in Mustache to represent conditions:
+
+```json
+{% raw %}{{#var}}var{{/var}}{% endraw %}
+```
+
+When `var` is a Boolean value, this syntax acts as an `if` condition. The `{% raw %}{{#var}}{% endraw %}` and `{% raw %}{{/var}}{% endraw %}` tags insert the values placed between them only if `var` evaluates to `true`.
+
+Using section tags would make your JSON invalid, so you must write your query in a string format instead.
+
+In this command, if you change the `limit` parameter to `true`, you would get back only two documents.
+That's because only when the condition is `true` the `size` parameter is activated.
+
+```json
+GET _search/template
+{
+  "source": "{% raw %}{ {{#limit}} \"size\": \"{{size}}\", {{/limit}}  \"query\":{\"match\":{\"play_name\": \"{{play_name}}\"}}}{% endraw %}",
+  "params": {
+    "play_name": "Henry IV",
+    "limit": false,
+    "size": 2
+  }
+}
+```
+
+You can also design an `if-else` condition.
+This command sets `size` to `2` if `limit` is `true` or sets it to `10` otherwise:
+
+```json
+GET _search/template
+{
+  "source": "{% raw %}{ {{#limit}} \"size\": \"2\", {{/limit}} {{^limit}} \"size\": \"10\", {{/limit}} \"query\":{\"match\":{\"play_name\": \"{{play_name}}\"}}}{% endraw %}",
+  "params": {
+    "play_name": "Henry IV",
+    "limit": true
+  }
+}
+```
+
+### Loops
+
+You can also use the section tag to implement a `foreach` loop:
```

Comment:
Unless we're using foreach as a keyword in a template or something, I think we can just leave it outside of the monospace:

to implement a foreach loop: